ISO images as storage volumes (#11852)
lxc/lxd: doc/howto/instances_create.md

    lxc launch images:ubuntu/22.04 my-instance --type t2.micro

The list of supported clouds and instance types can be found at [`https://github.com/dustinkirkland/instance-type`](https://github.com/dustinkirkland/instance-type).

### Launch a VM that boots from an ISO

> To launch a VM that boots from an ISO, you first need to create a VM. In this scenario it might make sense to create an empty VM as the ISO image could be used for installation. An empty VM can be created using the following command:

Reviewer: ```suggestion
To launch a VM that boots from an ISO, you must first create a VM. 
Let's assume that we want to create a VM and install it from the ISO image. 
In this scenario, use the following command to create an empty VM:
```

    lxc init iso-vm --empty --vm

> The second step it to import an ISO image which can later be attached to the VM as a storage volume:

Reviewer: ```suggestion
The second step is to import an ISO image that can later be attached to the VM as a storage volume:
```

>     lxc storage volume import path-to-image.iso iso-volume --type=iso

Reviewer: ```suggestion
    lxc storage volume import <path-to-image.iso> iso-volume --type=iso
```

Lastly, you need to attach the custom ISO volume to the VM using the following command:

    lxc storage volume attach <pool_name> iso-volume iso-vm

> Now it's time to start the VM. By default the VM will try booting from its disk. For the VM to boot from the attached ISO, you need to be connected to the console console, and repeatedly press `ESC` until the firmware menu appears:

Reviewer: ```suggestion
Now it's time to start the VM. 
By default, the VM will try booting from its disk. 
To make the VM boot from the attached ISO instead, connect to the console and repeatedly press `ESC` until the firmware menu appears:
```

Reviewer 2: @monstermunchkin have you tried making this easier by changing the boot.priority setting on the disk device when its attached by using `lxc config device add ` rather than `lxc storage volume attach`?

    lxc start iso-vm --console

Once you have entered the firmware menu, select `Boot Manager`, and `UEFI QEMU QEMU CD-ROM`. If you have multiple ISO images attached to the VM, the entries in the `Boot Manager` will be `UEFI QEMU QEMU CD-ROM`, `UEFI QEMU QEMU CD-ROM 2`, and so on.

The VM will now boot from the ISO image. You will likely need to disconnect from the serial console using `ctrl+a-q`, and connect to the VGA console using the following command:

>     lxc console iso-vm --type=vga

Reviewer 2: If you can add the iso volume as a `disk` device with a boot.priority setting to avoid needing to use the Boot manager above, then you can just do `lxc start iso-vm --console=vga`

Author: That's true. But then you'd have to remove the boot priority when you're done, which means you need to stop the VM, and remove `boot.priority` before starting it again.

 I could mention both ways of doing it.

Reviewer 2: Can they be hot removed?

Reviewer: This is an example - let's go for one way (ideally the easier one) and document that.
No need to cover all different ways of achieving the same thing.

Author: > Can they be hot removed?

They can, yes.

Author: @tomponline actually, we do need the serial console as there often is a menu where you need to select something or press any key (windows). Going straight to vga console therefore won't work.

Reviewer 2: ah OK.

> You should now see the installer. After the installation in done, the VM will reboot, and boot to disk without you needing to detach the custom ISO volume first.

Reviewer: ```suggestion
You should now see the installer. After the installation is done, the VM will reboot.
It will automatically boot to disk (you don't need to detach the custom ISO volume).
```

Reviewer 2: If we add the disk with boot.priority then we should instruct the user to remove the disk afterwards.
